docs/app-integration-development/19-OAuth-Functionality.md

```diff
@@ -0,0 +1,69 @@
+
+The access available to an application using Scoped OAuth differs by token type:
+* When using an act as app token, access is only limited by the scopes granted to the application. If the application has the `incidents.read` scope, it will be able to read all incidents on the account.
+* When using an act as user token, access is the intersection of the scopes granted to the application and the permissions of the user. If an application has the `incidents.read` scope and is acting as user Pagey, it will only be able to read incidents that user Pagey has permission to see.
+
+If a [REST API](/api-reference/) endpoint works with Scoped OAuth, the documentation for that endpoint will say "Scoped OAuth requires:" and list the required scope to access the endpoint.
+
+Scoped OAuth only works for private PagerDuty Apps at this time. The PagerDuty App is limited to accessing the PagerDuty Account that created it.
+
+### Understanding what type of OAuth functionality to use
+
+#### Scoped OAuth offers improved security
+Scoped OAuth improves on Classic User OAuth by offering a set of scopes that individually grant access to read or write a single type of resource. For example, the `incidents.read` scope is only able to read incidents on an account. The `incidents.write` scope is able to create, update, or delete an incident but is not able to read existing incidents.
+
+#### Scoped OAuth can act as an app or as a user
+Scoped OAuth has the ability to outside the context of a user as a PagerDuty App. This is useful in server to server flows where a user may not be involved. It also eliminates the need to manage service or bot user accounts. When configured with a redirect URI, Scoped OAuth applications can also obtain act as user tokens with the [Act as User Tokens via Code Grant](09-Act-as-User-OAuth-Code-Grant.md) flow to obtain a user's authorization and consent.
+
+#### Classic User OAuth is still the only supported mechanism for public apps
+If you are building an integration on top of PagerDuty with the intention of making it usable across many PagerDuty Accounts then Classic User OAuth continues to be the right choice.
+
+#### Scoped OAuth applications should be able to securely store the client secret
+All Scoped OAuth applications have a `client_id` and `client_secret` that can be used to obtain act as app tokens. Applications using Scoped OAuth functionality should have a server-side component where the `client_secret` is properly secured.
+
+#### Code Grant or PKCE?
+Scoped OAuth functionality is only able to obtain [Act as User Tokens via Code Grant](09-Act-as-User-OAuth-Code-Grant.md) at this time.
```

Comment:
I know this is about auth code vs auth code with PKCE, but it feels weird not to mention client credentials grant here.

Reply:
Yeah I didn't do that here because this primarily feels about "PKCE versus Code Grant" and the whole "public / confidential client" thing. I'm open to a separate "which flow should I use" question later...

Reply:
PKCE is still the Authorization Code flow though .. maybe it should be "To PKCE or not to PKCE?"

Reply:
Updated.